hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/DefaultNearCacheManager.java

```diff
@@ -107,5 +132,96 @@ public void destroyAllNearCaches() {
             nearCacheMap.remove(nearCache.getName());
             nearCache.destroy();
         }
+        for (ScheduledFuture preloadTaskFuture : preloadTaskFutures) {
+            preloadTaskFuture.cancel(true);
+        }
+
+        if (storageTaskFuture != null) {
+            storageTaskFuture.cancel(true);
+        }
+    }
+
+    private void createAndSchedulePreloadTask(NearCache nearCache, DataStructureAdapter adapter) {
+        if (adapter != null) {
+            PreloadTask preloadTask = new PreloadTask(nearCache, adapter);
+            ScheduledFuture<?> scheduledFuture = executionService.schedule(preloadTask, 3, SECONDS);
+            preloadTask.setScheduledFuture(scheduledFuture);
+
+            preloadTaskFutures.add(scheduledFuture);
+        }
+    }
+
+    private void createAndScheduleStorageTask() {
+        if (storageTaskFuture == null) {
+            StorageTask storageTask = new StorageTask();
+            storageTaskFuture = executionService.scheduleWithRepetition(storageTask, 0, 1, SECONDS);
+        }
+    }
+
+    private class PreloadTask implements Runnable {
+
+        private final NearCache nearCache;
+        private final DataStructureAdapter adapter;
+
+        private volatile ScheduledFuture<?> scheduledFuture;
+
+        PreloadTask(NearCache nearCache, DataStructureAdapter adapter) {
+            this.nearCache = nearCache;
+            this.adapter = adapter;
+        }
+
+        @Override
+        public void run() {
+            nearCache.preload(adapter);
+            if (scheduledFuture != null) {
+                preloadTaskFutures.remove(scheduledFuture);
+            }
+        }
+
+        void setScheduledFuture(ScheduledFuture<?> scheduledFuture) {
```

Comment:
What is the added value of this setter? Why not directly write to the scheduledFuture field?